src/main/java/com/gmail/goosius/siegewar/listeners/SiegeWarTownyEventListener.java

```diff
@@ -249,4 +261,157 @@ public void on (TownyFriendlyFireTestEvent event) {
             event.setPVP(true);
         }
     }
+
+    /**
+     * Broadcast rank removal in SiegeZones.
+     * 
+     * @param event TownRemoveResidentEvent.
+     */
+    @EventHandler(priority = EventPriority.MONITOR, ignoreCancelled = true)
+    public void on(TownRemoveResidentRankEvent event) {
+        tryBroadCastRankRemoval(event.getRank(), event.getResident());
+    }
+
+    /**
+     * Broadcast rank removal in SiegeZones.
+     * 
+     * @param event NationRankRemoveEvent
+     */
+    @EventHandler(priority = EventPriority.MONITOR, ignoreCancelled = true)
+    public void on(NationRankRemoveEvent event) {
+        tryBroadCastRankRemoval(event.getRank(), event.getResident());
+    }
+
+    /**
+     * Broadcast town removal in SiegeZones.
+     * 
+     * @param event TownRemoveResidentEvent.
+     */
+    @EventHandler(priority = EventPriority.MONITOR, ignoreCancelled = true)
+    public void on(TownRemoveResidentEvent event) {
+        tryBroadCastTownRemoval(event.getResident(), event.getTown());
+    }
+
+    /**
+     * Broadcast nation removal in SiegeZones.
+     * 
+     * @param event NationRemoveTownEvent.
+     */
+    @EventHandler(priority = EventPriority.MONITOR, ignoreCancelled = true)
+    public void on(NationRemoveTownEvent event) {
+        tryBroadCastNationRemoval(event.getTown(), event.getNation());
+    }
+
+    /**
+     * Broadcast nation ally removal in SiegeZones.
+     * 
+     * @param event NationRemoveAllyEvent.
+     */
+    @EventHandler(priority = EventPriority.MONITOR, ignoreCancelled = true)
+    public void on(NationRemoveAllyEvent event) {
+        tryBroadCastNationAllyRemoval(event.getRemovedNation(), event.getNation());
+    }
+
+    /**
+     * Check over the players in a nation when that nation has been removed as an
+     * ally, to see if they're in a siegezone.
+     * 
+     * @param removedNation Nation which is no longer an ally.
+     * @param nation        Nation which has removed an ally.
+     */
+    private void tryBroadCastNationAllyRemoval(Nation removedNation, Nation nation) {

```

Comment:
:construction: 
This methods needs to check the residents of both nations

because this event could be either
1. A nation which is an official attacker/defender, removing an ally.
2. A nation which is not an official attacker/defender, removing an ally which is an official attacker/defender.

Reply:
Found  to be not an issue, as Towny sends 2 events, one for each nation